internal/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,140 @@
+package healthcheck
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/chia-network/go-chia-libs/pkg/rpc"
+	"github.com/chia-network/go-chia-libs/pkg/types"
+)
+
+// Healthcheck is the main container for the app
+type Healthcheck struct {
+	healthcheckPort uint16
+	client          *rpc.Client
+
+	// Last block height we received
+	lastHeight uint32
+
+	// Time we received the last block height
+	lastHeightTime time.Time
+}
+
+// NewHealthcheck returns a new instance of healthcheck
+func NewHealthcheck(port uint16, logLevel log.Level) (*Healthcheck, error) {
+	var err error
+
+	healthcheck := &Healthcheck{
+		healthcheckPort: port,
+	}
+
+	log.SetLevel(logLevel)
+
+	healthcheck.client, err = rpc.NewClient(rpc.ConnectionModeWebsocket, rpc.WithAutoConfig(), rpc.WithBaseURL(&url.URL{
+		Scheme: "wss",
+		Host:   viper.GetString("hostname"),
+	}))
+	if err != nil {
+		return nil, err
+	}
+
+	return healthcheck, nil
+}
+
+// OpenWebsocket sets up the RPC client and subscribes to relevant topics
+func (h *Healthcheck) OpenWebsocket() error {
+	err := h.client.Subscribe("metrics")
+	if err != nil {
+		return err
+	}
+
+	err = h.client.AddHandler(h.websocketReceive)
+	if err != nil {
+		return err
+	}
+
+	h.client.AddDisconnectHandler(h.disconnectHandler)
+	h.client.AddReconnectHandler(h.reconnectHandler)
+
+	return nil
+}
+
+// CloseWebsocket closes the websocket connection
+func (h *Healthcheck) CloseWebsocket() error {
+	//return m.client.DaemonService.CloseConnection()
+	return nil
+}
+
+// StartServer starts the metrics server
+func (h *Healthcheck) StartServer() error {
+	log.Printf("Starting healthcheck server on port %d", h.healthcheckPort)
+
+	http.HandleFunc("/full_node", h.fullNodeHealthcheck())
+	return http.ListenAndServe(fmt.Sprintf(":%d", h.healthcheckPort), nil)
+}
+
+func (h *Healthcheck) websocketReceive(resp *types.WebsocketResponse, err error) {
```

Comment:
Maybe the logic in this function could be arranged in a way that would make it easier to contribute future healthcheck endpoints for other chia components? Perhaps some way of narrowing down what kind of websocket event was received, and by default doing nothing, but if it's an event that we want for an existing healthcheck, maybe call a `fullnodeWebsocketReceive` func or something like that? Not necessary advice, just seems like it would make a pattern that's easier for future contributors to extend this

Reply:
Yeah I mean, thats the pattern we have set up in chia-exporter (this all should look like a very stripped down version of that). But it seemed a bit silly to have a giant unused switch here